data/processing/input_latency.html

```diff
@@ -0,0 +1,319 @@
+            handle_portion.duration = handle_portion.end - latency_mid;
+
+            queue_uninteresting.push(queue_portion);
+            handle_uninteresting.push(handle_portion);
+            queue_active_uninteresting += queue_portion.duration;
+            handle_active_uninteresting += handle_portion.duration
+          }
+        }
+
+        // Add rendererIdle times, which are the same as the Bottom-Up approach
+        queue_uninteresting.push({"title":"rendererIdle", "duration":queue_renderer_idle});
+        handle_uninteresting.push({"title":"rendererIdle", "duration":handle_renderer_idle});
+        queue_active_uninteresting += queue_renderer_idle;
+        handle_active_uninteresting += handle_renderer_idle; 
+
+        // Calculate uninteresting time, which is all unaccounted for time (total time minus the 
+        // active time, which are the events we consider interesting plus the time spent idle, what
+        // remains is the time spent in uninteresting tasks) 
+        var queue_uninteresting_time = (latency_mid - latency_start) - queue_active_uninteresting;
+        queue_uninteresting.push({"title":"uninteresting", "duration":queue_uninteresting_time});        
+        queue_active_uninteresting += queue_uninteresting_time;
+        var handle_uninteresting_time = (latency_end - latency_mid) - handle_active_uninteresting;
+        handle_uninteresting.push({"title":"uninteresting", "duration":handle_uninteresting_time});
+        handle_active_uninteresting += handle_uninteresting_time;
+
+	results.push({"Latency Duration":latency_duration, "Queue Events Bottom-Up":mergeSameNameEvents(queue), 
+                      "Handling Events Bottom-Up":mergeSameNameEvents(handle), "Bottom-Up Event Sum":queue_active + handle_active, 
+                      "Queue Events Top-Down":mergeSameNameEvents(queue_uninteresting, "duration"), 
+                      "Handling Events Top-Down":mergeSameNameEvents(handle_uninteresting, "duration"), 
+                      "Top-Down Event Sum":queue_active_uninteresting + handle_active_uninteresting});
+      }
+    }
+    result.addPair("Latencies", results);
+  }
+
+  tr.mre.FunctionRegistry.register(inputLatencyMapper);
+
+  return {
+    inputLatencyMapper,
+  };
+
+  function getInputLatencyEvents(browserHelper) {
+    const isInputLatencyEvent = e => e.title === LATENCY_EVENT;
+    return browserHelper.getAllAsyncSlicesMatching(isInputLatencyEvent);
+  }
+
+  function getRendererEventsInRange(browserHelper, rendererID, range_of_interest) {
+    return browserHelper.getAllAsyncSlicesMatching(function(slice) {
+      return (slice.parentContainer.getProcess().pid === rendererID &&
+              slice.parentContainer.name === "CrRendererMain" && 
+              slice.category != "disabled-by-default-renderer.scheduler" &&
+              slice.category != "renderer.scheduler" &&
+              slice.category != "blink.animations,devtools.timeline,benchmark,rail" &&
```

Comment:
There are some meta slices associated with Animation and the Scheduler that are more like latencyInfo.Flow events, not exactly an event slice in the sense that its capturing a callback but that it has some information that we want to report. Animation and scheduling have lots of these events and they are tagged as being on the RendererMain thread but I filter them out because if I don't they end up eclipsing rendererIdle time and bubbling to the top as 'events' that take a long time even though they are not what we want to be looking at.